test/cat.jl

```diff
@@ -118,17 +109,8 @@ module TestCat
     dtc = DataTable(a = NullableArray([2, 3, 4]))
     dtd = DataTable(Any[2:4], [:a])
     dtab = vcat(dta, dtb)
-    dtac = vcat(dta, dtc)
```

Comment:
This fails due to a failure in `Base.vcat` not knowing how to concatenate `CategoricalVectors` and `NullableVectors`. The behavior can be restored with https://github.com/JuliaData/DataTables.jl/pull/30/files#diff-1924b206e8f0967bd42ab99a13efbd3bR106 but I think this is probably something that needs a true fix via https://github.com/JuliaLang/julia/pull/20815